server/plugin/command.go

```diff
@@ -394,12 +394,21 @@ func (p *Plugin) handleSettings(_ *plugin.Context, _ *model.CommandArgs, paramet
 	if setting != settingNotifications && setting != settingReminders {
 		return "Unknown setting."
 	}
-
 	strValue := parameters[1]
 	value := false
-	if strValue == settingOn {
+	switch {
+	case strValue == settingOn:
+		value = true
+	case strValue == settingOnChange:
+		if setting != "reminders" {
+			return "Invalid value. Accepted values are: \"on\" or \"off\"."
+		}
 		value = true
-	} else if strValue != settingOff {
+		userInfo.Settings.OnChange = true
+	case strValue != settingOff:
+		if setting == "reminders" {
+			return "Invalid value. Accepted values are: \"on\" or \"off\" or \"on-change\" ."
+		}
 		return "Invalid value. Accepted values are: \"on\" or \"off\"."
 	}
```

Comment:
This is becoming very hard to read. Let's just `switch` based on `setting`.

Reply:
resolved , changed to switch